model/assembly/CompiledTestConverterService.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoDeliveryRdf\model\assembly;
+
+use InvalidArgumentException;
+use tao_models_classes_service_StorageDirectory;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiTest\models\CompilationDataService;
+use oat\taoQtiTest\models\PhpCodeCompilationDataService;
+use oat\taoQtiTest\models\XmlCompilationDataService;
+
+class CompiledTestConverterService extends ConfigurableService
+{
+    const SERVICE_ID = 'taoDeliveryRdf/CompiledTestConverterService';
+
+    const OPTION_PHP_COMPILATION_SERVICE = 'php_compilation_service';

```

Comment:
the class name is converterServer, it is an abstraction, but then you have direct implementation: convert from php to xml, I guess you have to rename this class to make his destiny more clear

Reply:
It's not an abstraction. We have only two possible ways to store compiled test data - PHP and XML. Therefore I created one service with two methods to convert PHP->XML and XML->PHP. 
I didn't create two different services because they would have absolutely the same configuration and for me it seemed better to have one service with two different methods.